### PR TITLE
2936 debug docker build

### LINKED ISSUE #2936: Debug variant of docker builds.
### Description
As a client Operator, I want a variant docker image that includes troubleshooting and debugging tools so I can diagnose issues, observe the system, and submit more detailed bug reports.

### Acceptance Criteria
For at least point releases (maybe RCs too?) we would like a docker image, based on the existing openjdk one, but adds:

- full jdk, including jmd, jtop, jstat etc
- wget, curl, jq
- exposed debug port
- prescriptively enable DEBUG rpc api, debug on 5005, native memory tracking
- a working/updated apt repo

That way I can drop a new image into my existing container management system for use when debugging.

### PR DESCRIPTION
**Pull request by jflo**
## PR description
Adds a debug profile, and updates the latest profile to openjdk-17

## Fixed Issue(s)
fixes #2936 

## Changelog

- [X] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

**Comment by jflo** [comment]
fixes #2800

File: build.gradle [hunk 1]
```diff
@@ -614,7 +614,8 @@ artifactoryPublish {
 def dockerVariants = [
   "openjdk-11",
   "graalvm",
-  "openjdk-latest"
+  "openjdk-latest",
+  "openjdk-11-debug"
```
Comment (fab-10) [inline build.gradle:618]: just an esthetic suggestion, what about moving it just below "openjdk-11" to have a nicer grouping?

File: ethereum/p2p/build.gradle [hunk 2]
```diff
@@ -43,8 +43,12 @@ dependencies {
   implementation 'org.apache.logging.log4j:log4j-api'
   implementation 'org.apache.tuweni:tuweni-bytes'
   implementation 'org.apache.tuweni:tuweni-crypto'
-  implementation 'org.apache.tuweni:tuweni-devp2p'
```
Comment (fab-10) [inline ethereum/p2p/build.gradle:46]: This change seems to be independent, should it go in its own PR?

Comment (jflo) [reply ethereum/p2p/build.gradle:46]: It's a good question. I considered removing it, but since it solves for an existing open issue, i left it in and instead linked this PR to that issue. I think this falls under "clean as you go" but I admit that can be a subjective decision.

**Review by fab-10 — APPROVED** [review]
LGTM.

Nice to have, is a quick doc, with some examples, on how to interact with the debug container

**Review by macfarla — APPROVED** [review]
